internal/pkg/log/debug.go

```diff
@@ -12,20 +14,97 @@ import (
 // Logs are stored in a buffer by ioutil.Writer.
 type debugLogger struct {
 	*zapLogger
-	*ioutil.Writer
+	all         *ioutil.Writer
+	debug       *ioutil.Writer
+	info        *ioutil.Writer
+	warn        *ioutil.Writer
+	warnOrError *ioutil.Writer
+	error       *ioutil.Writer
+}
+
+// oneLevelEnabler enables only one level. The others are discarded.
+type oneLevelEnabler struct {
+	level zapcore.Level
+}
+
+func (v *oneLevelEnabler) Enabled(level zapcore.Level) bool {
+	return v.level == level
 }
 
 // NewDebugLogger returns logs as string by String() method.
 // See also other methods of the ioutil.Writer.
 func NewDebugLogger() DebugLogger {
-	writer := ioutil.NewBufferedWriter()
-	return &debugLogger{
-		zapLogger: loggerFromZap(zap.New(debugCore(writer))),
-		Writer:    writer,
+	l := &debugLogger{
+		all:         ioutil.NewBufferedWriter(),
+		debug:       ioutil.NewBufferedWriter(),
+		info:        ioutil.NewBufferedWriter(),
+		warn:        ioutil.NewBufferedWriter(),
+		warnOrError: ioutil.NewBufferedWriter(),
+		error:       ioutil.NewBufferedWriter(),
+	}
+	l.zapLogger = loggerFromZap(zap.New(zapcore.NewTee(
+		debugCore(l.all, DebugLevel),                            // all = debug level and higher
+		debugCore(l.debug, &oneLevelEnabler{level: DebugLevel}), // only debug msgs
+		debugCore(l.info, &oneLevelEnabler{level: InfoLevel}),   // only info msgs
+		debugCore(l.warn, &oneLevelEnabler{level: WarnLevel}),   // only warn msgs
+		debugCore(l.warnOrError, WarnLevel),                     // warn or error = warn level and higher
+		debugCore(l.error, ErrorLevel),                          // error = error level and higher
```

Comment:
`NewTee` zapisuje sa do vsetkych tychto destinacii.

`debugCore` - druhy parameter je `zapcore.LevelEnabler`.
Debug level, napr. `ErrorLevel` ho implementuje, a to tak, ze su povolene vsetky `levels` ako `error` a viac.
Preto som implementoval `oneLevelEnabler`, ktory povoli iba dany level, nie aj tie vyssie.